bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/SamsungTvTlsTrustManagerProvider.java

```diff
@@ -18,35 +18,52 @@
 
 import javax.net.ssl.SSLEngine;
 import javax.net.ssl.X509ExtendedTrustManager;
+import javax.security.auth.x500.X500Principal;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.smarthome.io.net.http.TlsTrustManagerProvider;
 import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Provides a TrustManager to allow secure websocket connections to any TV (=server)
+ * TODO: convert to using TrustAllTrustMananger (not supported in OH2.4 yet)
  *
  * @author Arjan Mels - Initial Contribution
  */
 @Component
 @NonNullByDefault
 public class SamsungTvTlsTrustManagerProvider implements TlsTrustManagerProvider {
+    private final static Logger logger = LoggerFactory.getLogger(WakeOnLanUtility.class);
+
     @Override
     public String getHostName() {
         return "SmartViewSDK";
     }
 
     private final X509ExtendedTrustManager trustAllCerts = new X509ExtendedTrustManager() {
 
+        private void logCertificateCN(String function, X509Certificate @Nullable [] x509Certificates) {
+            if (x509Certificates != null) {

```

Comment:
```suggestion
            if (x509Certificates != null && logger.isTraceEnabled()) {
```

Then we don't need to loop over all certificates and perform the getName if we don't log at all.

Reply:
Good idea.